frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx

```diff
@@ -52,18 +52,19 @@ const DeleteBlockPoolModal = withHandlePromise((props: DeleteBlockPoolModalProps
   );
   const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
 
+  // Check storage cluster is in ready state
+  const isClusterReady: boolean = cephCluster?.status?.phase === POOL_STATE.READY;
+
   React.useEffect(() => {
-    // restrict pool management for default pool and external cluster
+    // blocking pool management for default pool, external cluster and non-ready cluster
```

Comment:
all these three cases are considered as NOT ALLOWED, and if pool CR created but status is not ready then it will be considered as NOTREADY